### PR TITLE
[Backport 3.6] Add project and branch detection in shell

### PR DESCRIPTION
**Pull request by eleuzi01**
## Description

Resolves https://github.com/Mbed-TLS/mbedtls-framework/issues/39

## PR checklist

- [x] **changelog** not required, internal changes
- [x] **development PR** provided #9763 
- [x] **framework PR** provided Mbed-TLS/mbedtls-framework#67
- [x] **3.6 PR** provided
- [x] **2.28 PR** not required
- [x] **tests**  not required

### CODE REVIEW

**Comment by gilles-peskine-arm** [comment]
Note that there's a merge conflict with #9766 which is urgent because its framework side has already been merged. Please wait until #9766 is merged. (Strangely the main PR doesn't appear to have a conflict.)

**Review by mpg — APPROVED** [review]
LGTM

**Review by davidhorstmann-arm — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
